### PR TITLE
Convert rules svc to ext

### PR DESCRIPTION
**Pull request by adele-lc**
## Description of the change
New cli command and logic added that converts an org's `dr-general` and `dr-managed` rules using old service request calls into new extension requests

Syntax checks for rules verified in [notion doc](https://www.notion.so/Svc-ext-rules-8b1e189b43bc4a01b821d84b862d29aa)

Rules impacted:
- dumper
- pager-duty
- reliable-tasking
- velociraptor
- yara
- zeek

📷 Sample of cli response from Dry Run of command:
Screenshot 2024-02-28 9 19 13 PM

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: limacharlie/Extensions.py [hunk 1]
```diff
@@ -57,7 +42,72 @@ def delete( self, extName ):
...
+            if self._manager._jwt is None:
+                self._manager._refreshJWT()
+            req[ 'impersonator_jwt' ] = self._manager._jwt
+        return self._manager._apiCall( 'extension/request/%s' % ( extName, ), POST, req )
+    
+    def convert_rules(self, extName, isDryRun = True): 
+        extList = self.list()
+        if extName not in extList:
+            return f'ERROR: unable to convert rules. {self._manager._oid} is not subscribed to {extName}'
+        gen_hive = Hive.Hive(self._manager, "dr-general")
```
Comment (josh-lc) [inline limacharlie/Extensions.py:67]: Hive has no attribute Hive this should be Hive(self._manager, "dr-general")

Also did you have a chance to do a live run against some test data?

File: limacharlie/Extensions.py [hunk 2]
```diff
@@ -57,7 +42,72 @@ def delete( self, extName ):
...
+        man_hive = Hive.Hive(self._manager, "dr-managed")
```
Comment (josh-lc) [inline limacharlie/Extensions.py:68]: Hive has no attribute Hive this should be Hive(self._manager, "dr-general")

File: limacharlie/Extensions.py [hunk 3]
```diff
@@ -57,7 +42,72 @@ def delete( self, extName ):
...
+                print(f"Dry run of change on rule '{updated_rule['r_name']}':")
+                print("\033[91m- {}\033[0m".format(updated_rule['old_dnr'])) # print red text
+                print("\033[92m+ {}\033[0m".format(updated_rule['new_dnr'])) # print green text
+        if not isDryRun and len(updated_rules) > 0:
+            for updated_rule in updated_rules:
+                data = {
+                    "data": updated_rule['new_dnr']
+                }
+                try:
+                    hr = Hive.HiveRecord(updated_rule['r_name'], data)
```
Comment (josh-lc) [inline limacharlie/Extensions.py:102]: Same here Hive.HiveRecord does not have property HiveRecord 
You can do a import similar to that of Hive to bring HiveRecord into scope 
from limacharlie import HiveRecord


Then HiveRecord('','')

**Review by josh-lc — DISMISSED** [review]

File: limacharlie/Extensions.py (hunk 1 above)
Comment (adele-lc) [reply limacharlie/Extensions.py:67]: I did, yes. All test in cli and ui work. That just ended up being a naming convention from the import (line 2)
`from limacharlie import Hive`
it's grabbing attributes 'Hive' and 'HiveRecord' from the file 'Hive'

Comment (adele-lc) [reply limacharlie/Extensions.py:67]: So went back to try the individual imports and got a 'circular import' error:
`ImportError: cannot import name 'HiveRecord' from partially initialized module 'limacharlie' (most likely due to a circular import) (/home/adele/python-limacharlie/limacharlie/__init__.py)`

Happy to address this in another way if you have an alternative suggestion but `Hive.Hive` and `Hive.HiveRecord` seems to resolve circular import issue and makes the whole thing work 🤷

**Pull request merged**